### PR TITLE
Adds size limit to attachments for form submission API

### PR DESCRIPTION
**Pull request by ajeety4**
## Product Description

This adds size limit to attachments other than form xml to the submission API at the django layer. The size limit is 15MB per attachment (same as in the mobile app).

## Technical Summary

JIRA Ticket: [INDIV-101](https://dimagi-dev.atlassian.net/browse/INDIV-101)
Basic reason is to have some limit and 15MB was chosen as it is quite reasonable and already present at the mobile app.

### Safety story

Changes made are covered by the unit tests.

### Automated test coverage

New test cases were added for the changes made.

### QA Plan

Don't think QA is necessary for this change, however I am open for thoughts on this.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[INDIV-101]: https://dimagi-dev.atlassian.net/browse/INDIV-101?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
Should we also limit the total size / number of attachments?

**Comment by ajeety4** [comment]
> Should we also limit the total size / number of attachments?

That is a good point @snopoke 
Yes, there should be some hard limit at least on either of those. I think number of attachments would be a good option from UX's perspective.  
Any thoughts on actual limits here?
Also looks like a good point to raise on gtd-product.
Tagging Manish -  @mkangia

**Comment by snopoke** [comment]
> Should we also limit the total size / number of attachments?

I don't think this needs to hold up this PR. It can be done separately.

> Any thoughts on actual limits here?

Perhaps we should start by looking at existing data to see what the stats are for attachment count.

**Comment by ajeety4** [comment]
> > Should we also limit the total size / number of attachments?
> 
> I don't think this needs to hold up this PR. It can be done separately.
> 
> > Any thoughts on actual limits here?
> 
> Perhaps we should start by looking at existing data to see what the stats are for attachment count.

Great! I will let Manish have a look before getting this PR merged.

**Review by mkangia — APPROVED** [review]
minor feedback, nothing blocking

**Comment by mkangia** [comment]
> Don't think QA is necessary for this change, however I am open for thoughts on this.

@ajeety4 Agreed, QA isn't needed. Though good to test it once on staging, for both invalid/valid attachments

File: corehq/apps/receiverwrapper/tests/test_submissions.py
Comment (ajeety4) [reply corehq/apps/receiverwrapper/tests/test_submissions.py:175]: Good point! I considered this however ended up using the original settings since the file size is not that big i.e 15MB and should not impact performance much.

File: corehq/ex-submodules/couchforms/getters.py
Comment (ajeety4) [reply corehq/ex-submodules/couchforms/getters.py:57]: Thanks for catching the naming part!
As for check itself, the `if not <>` will work fine. I was explicitly checking for `False` as a good practice here, however digging more on this, not really required in this scenario and plus it is slower in performance.
https://switowski.com/blog/checking-for-true-or-false/#conclusions

File: corehq/apps/receiverwrapper/tests/test_submissions.py
Comment (mkangia) [inline corehq/apps/receiverwrapper/tests/test_submissions.py:176]: nit: Looks like this is now just getting used at one place later for status code, you could probably just get rid of it now.

**Review by mkangia — APPROVED** [review]
This is good to go!

**Pull request merged**
